### PR TITLE
add CODEOWNERS file

### PR DESCRIPTION
**Pull request by spengrah**
Sets up code ownership to explicitly and separately grant review authorities:

- Modules Registry Curators: the registry itself, i.e. the new submissions and changes to items on the registry
- Module Curation Managers: the curation *system*, i.e. changes to everything else in the repo

### CODE REVIEW

File: .github/CODEOWNERS
Comment (scottrepreneur) [inline .github/CODEOWNERS:4]: line 9 & 10

Comment (spengrah) [reply .github/CODEOWNERS]: good catch, thanks!

**Review by mulligan-madness — COMMENTED** [review]
If I understand correctly, this commit is just documenting in the repo the roles that curators and curation managers play in regards to different file paths? 

Curation managers, everything besides what is specified in lines 9 and 10, which is the modules directory and the modules.json file - owned exclusively by curators. Curators ONLY have ownership of the modules directory and modules.json.

Is there anything else mechanical going on here?

**Comment by spengrah** [comment]
> If I understand correctly, this commit is just documenting in the repo the roles that curators and curation managers play in regards to different file paths?
>
> Is there anything else mechanical going on here?

@mulligan-madness, good question! The answer, I'm pleased to say, is Yes. Github has a feature called Code Owners, which allows assignment of PR approval authority for specific files and directories to specific "owners." The way this works is via a combination of repo settings (configurable by a Haberdasher Labs admin, eg me) and the CODEOWNERS config file in the repo itself.

The result is that when changes are made to any of the files mapped to a given "owner," that owner — in our case the respective hat-based teams — are automatically assigned as reviewers with the requirement that 2+ of them approve the changes before they can be merged in.

> Curation managers, everything besides what is specified in lines 9 and 10, which is the modules directory and the modules.json file - owned exclusively by curators. Curators ONLY have ownership of the modules directory and modules.json.

The intent of this specific configuration here is to mirror as best as possible the authorities outlined in the respective hats. 

- Modules Registry Curators have authority to review/approve the registry content itself, which is manifested in the `modules/` directory and the `modules.json` file
- Module Curation Managers have authority to change the curation mechanism(s), which is defined by all the other files

**Review by gershido — COMMENTED** [review]
Looks great in general! 
There's an error in the CODEOWNERS file, the team name needs to include the org name: [`@org/team-name`](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#:~:text=%40username%20or-,%40org/team%2Dname,-format.%20Users%20and)

You can see the error on "Files changed" tab:
Screenshot 2024-01-06 at 8 51 39

**Review by gershido — APPROVED** [review]

**Review by scottrepreneur — APPROVED** [review]

**Pull request merged**
